### PR TITLE
Rare candy upgraded to use assimp.

### PR DESCRIPTION
**Pull request by Waterpicker**

### CODE REVIEW

**Review by Dysth — APPROVED** [review]

File: common/build.gradle.kts [hunk 1]
```diff
@@ -20,7 +20,8 @@ dependencies {
     implementation("gg.generations:RareCandy:${project.properties["rareCandy"]}"){isTransitive = false}
     implementation("org.tukaani:xz:${project.properties["rareCandyXZ"]}")
     implementation("org.apache.commons:commons-compress:${project.properties["rareCandyCommonCompress"]}")
-    implementation("de.javagl:jgltf-model:${project.properties["rareCandyJgltfModel"]}")
+    implementation("org.lwjgl:lwjgl-assimp:3.3.1")
```
Comment (JT122406) [inline common/build.gradle.kts:23]: are these needed I thought mc contained lejgl?

File: fabric/build.gradle.kts [hunk 2]
```diff
@@ -48,7 +49,8 @@ dependencies {
 
     implementation("shadowCommon"("gg.generations", "RareCandy", "${project.properties["rareCandy"]}"){isTransitive = false})!!
     implementation("shadowCommon"("org.tukaani", "xz", "${project.properties["rareCandyXZ"]}"))!!
-    implementation("shadowCommon"("de.javagl", "jgltf-model", "${project.properties["rareCandyJgltfModel"]}"))!!
+    implementation("shadowCommon"("org.lwjgl:lwjgl-assimp:3.3.1")!!)
```
Comment (JT122406) [inline fabric/build.gradle.kts:52]: see above

File: forge/build.gradle.kts [hunk 3]
```diff
@@ -58,7 +59,8 @@ dependencies {
 
     forgeRuntimeLibrary("shadowCommon"("gg.generations", "RareCandy", "${project.properties["rareCandy"]}"){isTransitive = false})!!
     forgeRuntimeLibrary("shadowCommon"("org.tukaani", "xz", "${project.properties["rareCandyXZ"]}"))!!
-    forgeRuntimeLibrary("shadowCommon"("de.javagl", "jgltf-model", "${project.properties["rareCandyJgltfModel"]}"))!!
+    forgeRuntimeLibrary("org.lwjgl:lwjgl-assimp:3.3.1")
```
Comment (JT122406) [inline forge/build.gradle.kts:62]: see above

File: gradle.properties [hunk 4]
```diff
@@ -24,7 +24,7 @@ parchment=2023.09.03
 
 badPackets=0.4.3
 WTHIT=8.6.0
-rareCandy=2.4.29-SNAPSHOT
+rareCandy=2.5.0-SNAPSHOT
```
Comment (JT122406) [inline gradle.properties:27]: if this is a major update we should bump rarecandy to 3.x.x

Comment (Waterpicker) [reply gradle.properties:27]: 3.0.0 is hydos's RKS branch. For I just switched to backporting the assimp code.

Comment (JT122406) [reply gradle.properties:27]: ok

**Comment by JT122406** [comment]
Does this include improvements to decrease file size?

**Comment by Waterpicker** [comment]
> Does this include improvements to decrease file size?
Yes. Assimp support the draco compression algorithm required to make smaller sizes. The other half was the improvements is already supported.

File: common/build.gradle.kts (hunk 1 above)
Comment (Waterpicker) [reply common/build.gradle.kts:23]: Assimp I don't think so.

**Review by JT122406 — APPROVED** [review]

**Pull request merged**